generators/csharp/codegen/src/ast/InstantiatedPrimitive.ts

```diff
@@ -115,14 +116,18 @@ export class PrimitiveInstantiation extends AstNode {
                 break;
             }
             case "dateTime": {
-                writer.write(`DateTime.Parse("${this.internalType.value.toISOString()}", null, DateTimeStyles.`);
-                writer.writeNode(
-                    csharp.classReference({
-                        name: "AdjustToUniversal",
-                        namespace: "System.Globalization"
-                    })
-                );
-                writer.write(")");
+                if (this.internalType.parsed) {
```

Comment:
nit: Should we call this `parse` since this is the case where we actually call `DateTime.Parse`? Otherwise it sounds like the string is already parsed and we do _not_ need to call `DateTime.Parse`.